test/e2e/receive_test.go

```diff
@@ -240,4 +267,55 @@ func TestReceive(t *testing.T) {
 			},
 		})
 	})
+
+	t.Run("multitenancy", func(t *testing.T) {
+		t.Parallel()
+
+		s, err := e2e.NewScenario("e2e_test_for_multitenancy")
+		testutil.Ok(t, err)
+		t.Cleanup(e2ethanos.CleanScenario(t, s))
+
+		// The replication suite creates a three-node hashring but one of the
+		// receivers is dead. In this case, replication should still
+		// succeed and the time series should be replicated to the other nodes.
+		r1, err := e2ethanos.NewReceiver(s.SharedDir(), s.NetworkName(), "1", 3)
+		testutil.Ok(t, err)
+		h := receive.HashringConfig{
+			Endpoints: []string{
+				r1.GRPCNetworkEndpointFor(s.NetworkName()),
+			},
+		}
+
+		// Recreate again, but with hashring config.
+		r1, err = e2ethanos.NewReceiver(s.SharedDir(), s.NetworkName(), "1", 3, h)
+		testutil.Ok(t, err)
+		testutil.Ok(t, s.StartAndWaitReady(r1))
+		testutil.Ok(t, err)
+		conf1 := ReverseProxyConfig{
+			tenant_id: "tenant-1",
+			port:      ":9097",
+			target:    e2ethanos.RemoteWriteEndpoint(r1.NetworkEndpoint(8080)),
+		}
+		q, err := e2ethanos.NewQuerier(s.SharedDir(), "1", []string{r1.GRPCNetworkEndpoint()}, nil, nil, "", "")
+		testutil.Ok(t, err)
+		testutil.Ok(t, s.StartAndWaitReady(q))
+		go generateProxy(conf1)
+		prom1, _, err := e2ethanos.NewPrometheus(s.SharedDir(), "1", defaultPromConfig("prom1", 0, "http://localhost:9097/api/v1/receive", ""), e2ethanos.DefaultPrometheusImage())
```

Comment:
I think what's happening is that because the Prometheus is started in its own docker container, "localhost:9097" is actually not the same as the reverse proxy started by this test.

Maybe @bwplotka or @pracucci have a suggestion how to do something like this best with the e2e test framework?

Reply:
As in other tests, you can use `r1.HTTPEndpoint()` to get actual address. (:

Reply:
Ok. I will change it to this. But the error which im facing is "No storeAPI matched for this query".

Reply:
Ok, I think I see the problem. We would need to inject ourselves into the docker network to inject  proxy server being run in unit test. I think it's doable, but it might simpler to create a container for this proxy first :hugs: Just creating a small service with container having such code like here would do there work: https://github.com/thanos-io/thanos/blob/17343c16b820a6fd5444e4258f4264ed1de065cc/test/e2e/e2ethanos/services.go#L117